Shell_Sort_and_Quick_Sort.cpp

```diff
@@ -0,0 +1,408 @@
+ 			int fNo = 0; 							// number of fields read so far
+ 			int pre = 0, pos = 0;					// locate one field of input record
+ 			temp.assign(rBuf); 						// copy string as a c-string
+ 			pos = temp.find_first_of('\t', pre); 	// locate the first field by '\t' (tab)
+ 			while (pos != string::npos) { 			// continue if it is not the end of record
+  				switch (++fNo) {
+  					case 1: oneC.cname = temp.substr(pre, pos - pre);
+ 						break; // get college name
+ 					case 2: oneC.dname = temp.substr(pre, pos - pre);
+ 						break; // get department name
+ 					case 3: oneC.level = temp.substr(pre, pos - pre);
+ 						break; // get level
+ 					default: break;
+ 				} //end switch
+ 				
+ 				pre = ++pos; 							// move to the next field
+ 				pos = temp.find_first_of('\t', pre); 	// locate the next delimiter
+ 			} //end inner-while
+ 			
+ 			pos = temp.find_last_of('\t');
+ 			temp = temp.substr(pos+1); 			// get total
+ 			oneC.total = atoi(temp.c_str()); 	// transform total into integer type
+ 			cSet.push_back(oneC); 				// save one record into the vector (array)
+ 		} //end outer-while
+ 	}	// else
+ 	
+ 	timer = clock();						// start the timer
+ 	timer = clock() - timer; 	// get the elapse time
+ 	cout << endl << "Ū�ɮɶ��G" << timer << " clocks (";
+ 	cout << ( (float)timer ) / CLOCKS_PER_SEC << " seconds)." << endl;
+ 	rTime = ( (float)timer ) / CLOCKS_PER_SEC ;
+	inFile.close(); // close file
+	fileName = temp;
+ 	if ( !cSet.size() )
+ 		cout << endl << "### Get nothing from the file " << fileName << " ! ### " << endl;
+} // end readRecords
+
+
+void displayAll(vector<collegeType> &cSet) { 	// output all on screen
+ 	int j = 0;
+ 	cout << endl << "*** There are " << cSet.size() << " records in total." << endl;
+ 	for (vector<collegeType>::iterator it = cSet.begin(); it < cSet.end(); ++it) {
+  		cout << "(" << setw(2) << ++j << ") " << setw(10) << it->cname << setw(30)
+			 << it->dname;
+ 		cout << setw(8) << it->level  setw(6) << it->total << " students" << endl;
+ 	} // end for
+} // end displayAll
+
+
+void writeRecords( vector<collegeType> &aSet, string fname, string fileName ) { // output all into a file
+ 	int j = 0;
+ 	fstream outFile; 								// output file handle
+ 	fname = "output" + fileName + fname + ".txt"; 	// output file name
+ 	outFile.open(fname.c_str(), fstream::out); 		// create a new file to write
+ 	if (!outFile.is_open()) { 						// unable to create a file
+ 		cout << endl << "### Cannot create " << fname << " ! ###" << endl;
+ 		return;
+ 	} //end if
+ 	
+ 	for (vector<collegeType>::iterator it = aSet.begin(); it < aSet.end(); ++it) {
+	 	outFile << "(" << setw(2) << ++j << ")" << "\t" << it->cname << "\t" 
+				<<	it->dname;
+ 		outFile << "\t" << it->level << "\t1234567" << it->total << endl;
+ 	} // end for
+ 	
+ 	outFile.close(); 	// close output file
+} // end writeRecords
+
+void WriteTimeRecords( string fileName, string fname, float readTime, float runTime ) {
```

Comment:
Subroutine/module mismatch (the header says we have a float-typed variable writeTime passed in as argument)